servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/HttpObjectDecoder.java

```diff
@@ -448,12 +452,16 @@ protected final void decodeLast(final ChannelHandlerContext ctx, final ByteBuf i
         // Handle the last unfinished message.
         if (message != null) {
             boolean chunked = isTransferEncodingChunked(message.headers());
-            if (currentState == State.READ_VARIABLE_LENGTH_CONTENT && !in.isReadable() && !chunked) {
-                // End of connection.
-                ctx.fireChannelRead(EmptyHttpHeaders.INSTANCE);
-                closeHandler.protocolPayloadEndInbound(ctx);
-                resetNow();
-                return;
+            if (!in.isReadable()) {
+                if ((currentState == State.READ_VARIABLE_LENGTH_CONTENT && !chunked) ||
+                        (currentState == State.READ_CHUNK_SIZE && chunked && !isDecodingRequest() &&
```

Comment:
Why is `isDecodingRequest()` a special case here?

Reply:
> On the server-side, this is not expected because if a client closes the connection server can not write the response:

Response may have already been sent.

If we follow spec for spec exceptions then isn't that ironic?

Reply:
> A client that sends a "close" connection option
   MUST close the connection after reading the final response message
   corresponding to this request.

> A server that sends a "close" connection option MUST initiate a close
   of the connection (see below) after it sends the response containing
   "close".

Server-side is a bit more involved. Because closure initiated by the server may be concurrent with closure initiated by the client, decoder can not correctly understand which side closes the connection. It requires more work and additional testing for this PR to make sure we do not behave incorrectly.

Let's limit the scope of this PR and defer server-side changes until we know the use-case.

Reply:
Discussed offline, we will make it work for client and server decoders. Tests for the server side will be added later.
Done in https://github.com/apple/servicetalk/pull/1060/commits/64fb4cc1ab4a8c3316a094560bd531dd236615f3